src/facade/dragonfly_connection.cc

```diff
@@ -613,10 +613,25 @@ void Connection::HandleRequests() {
     if (!(IsPrivileged() && no_tls_on_admin_port)) {
       // Must be done atomically before the premption point in Accept so that at any
       // point in time, the socket_ is defined.
+      uint8_t buf[2];
+      auto read_sz = socket_->Read(io::MutableBytes(buf));
+      if (!read_sz || *read_sz < sizeof(buf)) {
+        VLOG(1) << "Error reading from peer " << remote_ep << " " << read_sz.error().message();
+        return;
+      }
+      if (buf[0] != 0x16 || buf[1] != 0x03) {
```

Comment:
The former (`buf[0]`) should denote the `record type` and the later should be the `major` version of the protocol (in this case `3`). Two things here:

1. Do we want to deprecate earlier (older) `tls` versions? For example, if `buf[2]` (the minor version) is `0x1` it would mean that the TLS version is `1.0` which is deprecated since 2021 I think. Is this something that we care?

2. It happened to remember what those two bytes denote. However, for somebody who is just looking at the code it won't make much sense without googling. So I would suggest a small nit and replace the `0x16` and `0x03` with two variables named `tls_record_type` and `tls_major_version` respectively (or anything that clearly describes the context of those two bytes)

Reply:
I just want to differentiate between SSL/not SSL protocols and reply in plain text for the latter. All the other restrictions are encoded inside `CreateSslServerCntx()` function